### YamlMime:ManagedReference
items:
- uid: System.Threading.ThreadInterruptedException
  id: ThreadInterruptedException
  children:
  - System.Threading.ThreadInterruptedException.#ctor
  - System.Threading.ThreadInterruptedException.#ctor(System.String)
  - System.Threading.ThreadInterruptedException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.Threading.ThreadInterruptedException.#ctor(System.String,System.Exception)
  langs:
  - csharp
  name: ThreadInterruptedException
  nameWithType: ThreadInterruptedException
  fullName: System.Threading.ThreadInterruptedException
  type: Class
  summary: "ときにスローされる例外を<xref href=&quot;System.Threading.Thread&quot;></xref>待機状態にある間に中断します。"
  remarks: "1 つまたは複数のスレッドを作成すると後は、<xref:System.Threading.ThreadState>が破棄されるまでの状態します</xref:System.Threading.ThreadState>。 呼び出す<xref:System.Threading.Thread.Interrupt%2A>でスレッドがある場合、<xref:System.Threading.ThreadState>状態と、対象のスレッドでスローされる ThreadInterruptedException</xref:System.Threading.ThreadState> </xref:System.Threading.Thread.Interrupt%2A> 。 スレッドに含まれていない場合、<xref:System.Threading.ThreadState>状態にある場合、スレッドがその状態になるまで、例外はスローされません</xref:System.Threading.ThreadState>。 スレッドがブロックされない場合は、中断されることがなく完了でした。       ThreadInterruptedException は、値は 0x80131519 HRESULT COR_E_THREADINTERRUPTED を使用します。       ThreadInterruptedException のインスタンスの初期プロパティ値の一覧は、次を参照してください、<xref:System.Threading.ThreadInterruptedException.%23ctor%2A>コンス トラクター。</xref:System.Threading.ThreadInterruptedException.%23ctor%2A> 。"
  example:
  - "The following code example shows the behavior of a running thread when it is interrupted and subsequently gets blocked.  \n  \n [!code-cpp[System.Threading.Thread.Interrupt#1](~/add/codesnippet/cpp/t-system.threading.threa_1_1.cpp)]\n [!code-vb[System.Threading.Thread.Interrupt#1](~/add/codesnippet/visualbasic/t-system.threading.threa_1_1.vb)]\n [!code-cs[System.Threading.Thread.Interrupt#1](~/add/codesnippet/csharp/t-system.threading.threa_1_1.cs)]"
  syntax:
    content: >-
      [System.Runtime.InteropServices.ComVisible(true)]

      public class ThreadInterruptedException : SystemException
  inheritance:
  - System.Exception
  - System.SystemException
  implements: []
  inheritedMembers: []
  platform:
  - net462
- uid: System.Threading.ThreadInterruptedException.#ctor
  id: '#ctor'
  parent: System.Threading.ThreadInterruptedException
  langs:
  - csharp
  name: ThreadInterruptedException()
  nameWithType: ThreadInterruptedException.ThreadInterruptedException()
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException()
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "新しいインスタンスを初期化、 <xref href=&quot;System.Threading.ThreadInterruptedException&quot;> </xref>既定のプロパティを持つクラス。"
  remarks: "次の表は、 <xref:System.Threading.ThreadInterruptedException>。</xref:System.Threading.ThreadInterruptedException>のインスタンスの初期プロパティ値を示しています。      |プロパティ |値 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|`null`.|  |<xref:System.Exception.Message%2A>|ローカライズされたエラー メッセージ文字列 |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A> 。"
  syntax:
    content: public ThreadInterruptedException ();
    parameters: []
  overload: System.Threading.ThreadInterruptedException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Threading.ThreadInterruptedException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.Threading.ThreadInterruptedException
  langs:
  - csharp
  name: ThreadInterruptedException(String)
  nameWithType: ThreadInterruptedException.ThreadInterruptedException(String)
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException(String)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "新しいインスタンスを初期化、 <xref href=&quot;System.Threading.ThreadInterruptedException&quot;> </xref>指定したエラー メッセージを使用します。"
  remarks: "次の表は、 <xref:System.Threading.ThreadInterruptedException>。</xref:System.Threading.ThreadInterruptedException>のインスタンスの初期プロパティ値を示しています。      |プロパティ |値 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|`null`.|  |<xref:System.Exception.Message%2A>|エラー メッセージ文字列 |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A> 。"
  syntax:
    content: public ThreadInterruptedException (string message);
    parameters:
    - id: message
      type: System.String
      description: "例外の原因を説明するエラー メッセージ。"
  overload: System.Threading.ThreadInterruptedException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Threading.ThreadInterruptedException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  id: '#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)'
  parent: System.Threading.ThreadInterruptedException
  langs:
  - csharp
  name: ThreadInterruptedException(SerializationInfo,StreamingContext)
  nameWithType: ThreadInterruptedException.ThreadInterruptedException(SerializationInfo,StreamingContext)
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException(SerializationInfo,StreamingContext)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "新しいインスタンスを初期化、 <xref href=&quot;System.Threading.ThreadInterruptedException&quot;> </xref>シリアル化されたデータを持つクラス。"
  remarks: "このコンス トラクターは、ストリームで送信された例外オブジェクトを再構成する逆シリアル化中に呼び出されます。 詳細については、次を参照してください。 [XML および SOAP シリアル化](~/add/includes/ajax-current-ext-md.md)です。"
  syntax:
    content: protected ThreadInterruptedException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);
    parameters:
    - id: info
      type: System.Runtime.Serialization.SerializationInfo
      description: "<xref href=&quot;System.Runtime.Serialization.SerializationInfo&quot;> </xref>スローされた例外に関するシリアル化されたオブジェクト データを保持します。"
    - id: context
      type: System.Runtime.Serialization.StreamingContext
      description: "<xref:System.Runtime.Serialization.StreamingContext>ソースまたは転送先に関するコンテキスト情報を含む</xref:System.Runtime.Serialization.StreamingContext>。"
  overload: System.Threading.ThreadInterruptedException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Threading.ThreadInterruptedException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.Threading.ThreadInterruptedException
  langs:
  - csharp
  name: ThreadInterruptedException(String,Exception)
  nameWithType: ThreadInterruptedException.ThreadInterruptedException(String,Exception)
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException(String,Exception)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "新しいインスタンスを初期化、 <xref href=&quot;System.Threading.ThreadInterruptedException&quot;> </xref>指定したエラー メッセージおよびこの例外の原因となった内部例外への参照を持つクラス。"
  remarks: "前の例外への参照を含める必要があります前の例外の直接の結果としてスローされる例外、<xref:System.Exception.InnerException%2A>プロパティ</xref:System.Exception.InnerException%2A>。 <xref:System.Exception.InnerException%2A>プロパティは、コンス トラクターに渡されたものと同じ値を返しますまたは`null`場合、<xref:System.Exception.InnerException%2A>プロパティがコンス トラクターには、内部例外値を指定していません</xref:System.Exception.InnerException%2A></xref:System.Exception.InnerException%2A>。       次の表は、 <xref:System.Threading.ThreadInterruptedException>。</xref:System.Threading.ThreadInterruptedException>のインスタンスの初期プロパティ値を示しています。      |プロパティ |値 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|内部例外の参照 |。  |<xref:System.Exception.Message%2A>|エラー メッセージ文字列 |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A> 。"
  syntax:
    content: public ThreadInterruptedException (string message, Exception innerException);
    parameters:
    - id: message
      type: System.String
      description: "例外の原因を説明するエラー メッセージ。"
    - id: innerException
      type: System.Exception
      description: "現在の例外の原因となった例外。 場合、`innerException`パラメーターではありません<xref uid=&quot;langword_csharp_null&quot; name=&quot;null&quot; href=&quot;&quot;></xref>で現在の例外が発生、 <xref uid=&quot;langword_csharp_catch&quot; name=&quot;catch&quot; href=&quot;&quot;> </xref>ブロックは、内部例外を処理します。"
  overload: System.Threading.ThreadInterruptedException.#ctor*
  exceptions: []
  platform:
  - net462
references:
- uid: System.SystemException
  isExternal: false
  name: System.SystemException
- uid: System.Threading.ThreadInterruptedException.#ctor
  parent: System.Threading.ThreadInterruptedException
  isExternal: false
  name: ThreadInterruptedException()
  nameWithType: ThreadInterruptedException.ThreadInterruptedException()
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException()
- uid: System.Threading.ThreadInterruptedException.#ctor(System.String)
  parent: System.Threading.ThreadInterruptedException
  isExternal: false
  name: ThreadInterruptedException(String)
  nameWithType: ThreadInterruptedException.ThreadInterruptedException(String)
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException(String)
- uid: System.String
  parent: System
  isExternal: true
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.Threading.ThreadInterruptedException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.Threading.ThreadInterruptedException
  isExternal: false
  name: ThreadInterruptedException(SerializationInfo,StreamingContext)
  nameWithType: ThreadInterruptedException.ThreadInterruptedException(SerializationInfo,StreamingContext)
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException(SerializationInfo,StreamingContext)
- uid: System.Runtime.Serialization.SerializationInfo
  parent: System.Runtime.Serialization
  isExternal: false
  name: SerializationInfo
  nameWithType: SerializationInfo
  fullName: System.Runtime.Serialization.SerializationInfo
- uid: System.Runtime.Serialization.StreamingContext
  parent: System.Runtime.Serialization
  isExternal: true
  name: StreamingContext
  nameWithType: StreamingContext
  fullName: System.Runtime.Serialization.StreamingContext
- uid: System.Threading.ThreadInterruptedException.#ctor(System.String,System.Exception)
  parent: System.Threading.ThreadInterruptedException
  isExternal: false
  name: ThreadInterruptedException(String,Exception)
  nameWithType: ThreadInterruptedException.ThreadInterruptedException(String,Exception)
  fullName: System.Threading.ThreadInterruptedException.ThreadInterruptedException(String,Exception)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.Threading.ThreadInterruptedException.#ctor*
  parent: System.Threading.ThreadInterruptedException
  isExternal: false
  name: ThreadInterruptedException
  nameWithType: ThreadInterruptedException.ThreadInterruptedException
